pkg/objectstate/merge/merge.go

```diff
@@ -32,11 +32,11 @@ var (
 func ServiceAccountForUpdate(current, updated client.Object) (client.Object, error) {
 	curSA, ok := current.(*corev1.ServiceAccount)
 	if !ok {
-		return updated, ErrWrongObjectType
+		return nil, ErrWrongObjectType
```

Comment:
we should ignore the other return value anyway if `err != nil` , but using an explicit value (`nil`) is a tad better anyway